In [77]:
import numpy as np
import pandas as pd
import statistics as stat

In [78]:
df = pd.DataFrame()

In [79]:
df['name'] = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Oliver']
df['age'] = [14, 12, 11, 10, 8, 6, 8]

In [80]:
age_mean = np.mean(df['age'])
age_median = np.median(df['age'])
age_mode = stat.mode(df['age'])

In [81]:
print('The mean of kids ages is {}'.format(age_mean))
print('The median of kids ages is {}'.format(age_median))
print('The mode of kids ages is {}'.format(age_mode))

The mean of kids ages is 9.857142857142858
The median of kids ages is 10.0
The mode of kids ages is 8


In [82]:
age_var = np.var(df['age'])
age_std = np.std(df['age'], ddof=1)
age_error_mean = age_std / ((len(df['age']))**0.5) 

In [83]:
print('The variance of kids ages is {}'.format(age_var))
print('The standard deviation of kids ages is {}'.format(age_std))
print('The error of kids ages is {}'.format(age_error_mean))

The variance of kids ages is 6.408163265306122
The standard deviation of kids ages is 2.734262327610589
The error of kids ages is 1.0334540197243192


## 2. Choose one central tendency & one variance

    I will choose mean as estimated central tendency,      age_mean = 9.857
    The variance will be as the variance calculated above,  age_variance = 6.408
    
    The reason is that the data set is kind distributed around 9.8 and has no big low_end 
    or high_end tails.


## 3. after Cindy has a birthday

In [101]:
df.loc[df[df.name == 'Cindy'].index, 'age'] += 1

In [102]:
age_mean = np.mean(df['age'])
age_median = np.median(df['age'])
age_mode = stat.mode(df['age'])

In [103]:
print('The mean of kids ages is {}'.format(age_mean))
print('The median of kids ages is {}'.format(age_median))
print('The mode of kids ages is {}'.format(age_mode))

The mean of kids ages is 9.142857142857142
The median of kids ages is 10.0
The mode of kids ages is 8


In [104]:
age_var = np.var(df['age'])
age_std = np.std(df['age'], ddof=1)
age_error_mean = age_std / ((len(df['age']))**0.5)

In [105]:
print('The variance of kids ages is {}'.format(age_var))
print('The standard deviation of kids ages is {}'.format(age_std))
print('The error of kids ages is {}'.format(age_error_mean))

The variance of kids ages is 14.979591836734695
The standard deviation of kids ages is 4.180453381654971
The error of kids ages is 1.5800628593368629


# What have not changed are  median,mode
# 
# What have changed are  mean, variance, Std, error on mean

## 4. After replace Oliver by Jessica

In [106]:
df.loc[df[df.name == 'Oliver'].index, 'name'] = 'Jessica'
df.loc[df[df.name == 'Jessica'].index, 'age'] = 1

In [107]:
df.head(10)

name  age
0     Greg   14
1   Marcia   12
2    Peter   11
3      Jan   10
4    Bobby    8
5    Cindy    8
6  Jessica    1

In [108]:
age_mean = np.mean(df['age'])
age_median = np.median(df['age'])
#age_mode = stat.mode(df['age'])

## calculate multiple modes

In [109]:
(values, counts) = np.unique(df['age'], return_counts=True)
if len(counts) == len(df['age']):
    age_mode = age_median


In [110]:
print('The mean of kids ages is {}'.format(age_mean))
print('The median of kids ages is {}'.format(age_median))
print('The mode of kids ages is {}'.format(age_mode))

The mean of kids ages is 9.142857142857142
The median of kids ages is 10.0
The mode of kids ages is 8


In [111]:
age_var = np.var(df['age'])
age_std = np.std(df['age'], ddof=1)
age_error_mean = age_std / ((len(df['age']))**0.5)

In [112]:
print('The variance of kids ages is {}'.format(age_var))
print('The standard deviation of kids ages is {}'.format(age_std))
print('The error of kids ages is {}'.format(age_error_mean))

The variance of kids ages is 14.979591836734695
The standard deviation of kids ages is 4.180453381654971
The error of kids ages is 1.5800628593368629


## 5. Magazine survey

In [113]:
df_fans = pd.DataFrame()
df_fans['survey'] = ['TV Guide', 'Entertainment Weekly', 'Pop Culture Today', 'SciPhi Phanatic']
df_fans['percentage'] = [20, 23, 17, 5]

In [114]:
df_fans.head()

survey  percentage
0              TV Guide          20
1  Entertainment Weekly          23
2     Pop Culture Today          17
3       SciPhi Phanatic           5

In [115]:
aver = np.mean(df_fans['percentage'])
print(aver)

16.25


In [116]:
print('16.25% of adults were fans')

16.25% of adults were fans
